eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -32,6 +51,11 @@ jobs:
         OSVmImage: 'ubuntu-18.04'
         vcpkg.deps: ''
         AZ_SDK_C_NO_SAMPLES: 'true'
+      Linux_x64_gcc5:
+        OSVmImage: 'ubuntu-16.04'
+        vcpkg.deps: ''
+        AZ_SDK_C_NO_SAMPLES: 'true'
```

Comment:
If we are only going to have one leg, I would prefer it with samples being built.

Also, add a comment like the rest for the various flags on/off.

Reply:
I wanted to build with samples too but the samples require vcpkg dependencies. To build vcpkg you need g++ >= 6 ([link](https://github.com/Microsoft/vcpkg#quick-start-unix)).

Reply:
cc @BillyONeal